File: Python/SumGame.py
class Solution:
    def sumGame(self, num:str) -> bool:
        half = len(num)/2

        leftSum = 0
        rightSum = 0
        leftQuestionMark = 0
        rightQuestionMark = 0

        for i in range(0, len(num)):
            if num[i] == "?":
                if i<half:
                    leftQuestionMark += 1
                else: 
                    rightQuestionMark += 1

            else:
                if i<half: 
                    leftSum += int(num[i])
                else : rightSum += int(num[i])


        sumDiff = leftSum - rightSum
        questionDiff = leftQuestionMark - rightQuestionMark

        return sumDiff * 2 != -9*questionDiff

File: JavaScript/SumGame.js
const sumGame = function (num) {
  const half = num.length / 2;

  let leftSum = 0;
  let rightSum = 0;
  let leftQuestionMark = 0;
  let rightQuestionMark = 0;

  for (let i = 0; i < num.length; i++) {
    if (num[i] === "?") {
      if (i < half) leftQuestionMark++;
      else rightQuestionMark++;
    } else {
      if (i < half) leftSum += Number(num[i]);
      else rightSum += Number(num[i]);
    }
  }

  const sumDiff = leftSum - rightSum;
  const questionDiff = leftQuestionMark - rightQuestionMark;

  return sumDiff * 2 !== -9 * questionDiff;
};

sumGame("?3295???");
